### PR TITLE
Shop-manager-challenge

### PR DESCRIPTION
**Pull request by rkirkbride13**
My name:
Robbie Kirkbride

User stories:
- [X] User story 1: "I want to keep a list of my shop items with their name and unit price"
- [X] User story 2: "I want to know which quantity (a number) I have for each item"
- [X] User story 3: "I want to be able to create a new item"
- [X] User story 4: "I want to keep a list of orders with their customer name"
- [X] User story 5: "I want to assign each order to their corresponding item"
- [X] User story 6: "I want to know on which date an order was placed"
- [X] User story 7: "I want to be able to create a new order"

### CODE REVIEW

File: lib/item_repository.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  def create(item)
+    sql = 'INSERT INTO items (name, price, quantity) VALUES ($1, $2, $3);'
+    params = [item.name, item.price, item.quantity]
+    DatabaseConnection.exec_params(sql, params)
+    return nil
+  end
+
+  def delete(id)
```
Comment (shussain894) [inline lib/item_repository.rb:31]: good display of delete method but i dont think we required it from the user story, same with find the method

File: lib/item_repository.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  def update(item)
+    sql = 'UPDATE items SET name = $1, price = $2, quantity = $3 WHERE id = $4;'
+    params = [item.name, item.price, item.quantity, item.id]
+    DatabaseConnection.exec_params(sql, params)
+    return nil
+  end
+
+  private
+
+  def unpack_record(record)
```
Comment (shussain894) [inline lib/item_repository.rb:47]: clean use of a method for the records, i know its me being picky and private methods should go at the end but reading from the top i was unsure of what unpack_record meant or does, so maybe a comment at the top to allude to the fact that the method is private and at the bottom

File: spec/app_spec.rb [hunk 3]
```diff
@@ -0,0 +1,117 @@
+require_relative '../app'
+
+RSpec.describe Application do
```
Comment (shussain894) [inline spec/app_spec.rb:3]: something minor but have the tests in order of the shop program

File: spec/app_spec.rb [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    app = Application.new("shop_inventory_test", terminal, item_repository, order_repository)
+    app.run
+  end
+
+  it "creates an order if user input is 4" do
+    terminal = double :terminal
+    item1 = double :item, id: 1, name: "TV", price: 99.99, quantity: "5"
+    item_repository = double :item_repository, find: item1
+    order4 = double :order, id: 4, customer: 'Shah', date: '2022-10-10'
+    order_repository = double :order_repository, create: order4
```
Comment (marinapapap) [inline spec/app_spec.rb:97]: Great use of doubles to test that the application was appropriately using the create methods.

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    case input
+      when "1" then list_items
+      when "2" then create_item
+      when "3" then list_orders
+      when "4" then create_order
+      else @io.puts "Request not recognized"
+    end
+  end
+
+  private
```
Comment (marinapapap) [inline app.rb:29]: Great use of private methods to keep the code readable and DRY.

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  def create_order
+    order = Order.new
+    @io.puts "Customer name:"
+    order.customer = @io.gets.chomp
+    @io.puts "Order date:"
+    order.date = @io.gets.chomp
+    @io.puts "Item ordered:"
+    item_id = @io.gets.chomp
+    item = @item_repository.find(item_id)
+    @order_repository.create(order, item)
```
Comment (marinapapap) [inline app.rb:84]: A practical addition to the app class in linking the order to a new item. Next time, you can find a way to add multiple items to an order at one time.

**Review by marinapapap — COMMENTED** [review]
You've covered all the requirements and have refactored your code to be as readable and DRY as is possible. You've used doubles extensively throughout to unit test your application class and your test coverage seems to be close to 100%. Great work!

**Pull request closed without merging**